components/core/src/clp/streaming_archive/reader/File.cpp

```diff
@@ -252,14 +252,20 @@ bool File::find_message_in_time_range(
 SubQuery const* File::find_message_matching_query(Query const& query, Message& msg) {
     SubQuery const* matching_sub_query = nullptr;
     while (m_msgs_ix < m_num_messages && nullptr == matching_sub_query) {
-        auto logtype_id = m_logtypes[m_msgs_ix];
+        auto const curr_msg_ix{m_msgs_ix};
+        auto logtype_id = m_logtypes[curr_msg_ix];
 
         // Get number of variables in logtype
         auto const& logtype_dictionary_entry = m_archive_logtype_dict->get_entry(logtype_id);
         auto const num_vars = logtype_dictionary_entry.get_num_variables();
-
+        auto const var_begin_ix{m_variables_ix};
```

Comment:
```suggestion

        auto const var_begin_ix{m_variables_ix};
```

The comment above is not related to these two lines, so I think it's worth breaking up the two blocks.